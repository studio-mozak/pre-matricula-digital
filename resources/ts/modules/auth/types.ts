export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
