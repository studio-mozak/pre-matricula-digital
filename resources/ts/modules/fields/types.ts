import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
