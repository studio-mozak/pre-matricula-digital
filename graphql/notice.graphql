extend type Query @guard(with: ["web", "sanctum", "prematricula"]) {

  """
  Lista todos os avisos.
  """
  notices: [Notice!]! @paginate

  notice(id: ID @eq): Notice @find

}

extend type Mutation @guard {

  saveNotice(
    input: NoticeInput! @spread
  ): Notice!

  deleteNotice(
    id: ID! @whereKey
  ): Notice! @delete
}

type Notice {
  id: ID!
  text: String!
}

input NoticeInput {
  id: ID
  text: String!
}
