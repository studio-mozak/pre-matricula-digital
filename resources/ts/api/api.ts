import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
