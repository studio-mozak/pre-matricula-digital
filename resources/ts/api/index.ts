export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
