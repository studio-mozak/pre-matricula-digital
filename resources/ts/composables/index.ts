export * from './useLoader';
export * from './useLoaderAndShowErrors';
export * from './usePortal';
export * from './useModal';
export * from './useStudentProcessAndSuggestGrades';
