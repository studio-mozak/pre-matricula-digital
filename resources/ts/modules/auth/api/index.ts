/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
