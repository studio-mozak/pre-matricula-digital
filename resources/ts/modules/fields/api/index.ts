/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
