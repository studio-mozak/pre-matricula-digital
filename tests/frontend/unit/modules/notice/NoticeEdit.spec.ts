import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { Notice } from '@/modules/notice/types';
import { Notice as NoticeApi } from '@/modules/notice/api';
import NoticeEdit from '@/modules/notice/components/NoticeEdit.vue';
import VTooltip from 'v-tooltip';
import { expect, vi } from 'vitest';
import { useModal } from '@/composables';

const dialogSpy = vi.fn();

vi.mock('@/composables/useModal');
vi.mock('@/modules/notice/api');

const notice: Notice = {
  id: 1,
  text: 'test notice',
};

const onMounted = async (response: Notice) => {
  vi.mocked(NoticeApi.list).mockReturnValueOnce(Promise.resolve(response));

  vi.mocked(useModal).mockImplementation(() => ({
    dialog: dialogSpy,
  }));

  const wrapper = mount(NoticeEdit, {
    global: {
      directives: {
        tooltip: VTooltip,
      },
      stubs: ['x-field', 'x-modal'],
    },
  });

  await flushPromises();

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('NoticeEdit', () => {
  let wrapper: VueWrapper<InstanceType<typeof NoticeEdit>>;

  beforeEach(async () => {
    wrapper = await onMounted(notice);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.resetAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('notice value is successfully loaded', () => {
    expect(wrapper.vm.notice).toEqual(notice);
  });

  test('the value of notice is printed in the DOM', () => {
    const noticeDOM = wrapper.find('[type="RICH_TEXT"]');
    expect(noticeDOM.html()).toContain(notice.text);
  });

  test('notice value is empty', async () => {
    wrapper = await onMounted(undefined as unknown as Notice);

    expect(wrapper.vm.notice).toEqual({
      id: null,
      text: null,
    });
  });

  test('ao remover aviso, retorna erro', async () => {
    vi.mocked(NoticeApi.remove).mockRejectedValueOnce({});

    wrapper.vm.notice.text = '';

    const button = wrapper.find('[type="button"]');

    button.trigger('click');

    await flushPromises();
    await wrapper.vm.$nextTick();

    expect(dialogSpy).toHaveBeenCalledWith({
      title: 'Erro',
      description:
        'Não foi possível salvar o aviso. Por favor, entre em contato com o suporte.',
      titleClass: 'danger',
      iconLeft: 'status-red',
    });
  });

  test('remove aviso do sistema com sucesso', async () => {
    vi.mocked(NoticeApi.remove).mockReturnValueOnce(
      Promise.resolve({
        errors: false,
      })
    );

    wrapper.vm.notice.text = '';

    const button = wrapper.find('[type="button"]');

    button.trigger('click');

    await wrapper.vm.$nextTick();

    await flushPromises();

    expect(dialogSpy).toHaveBeenCalledWith({
      title: 'Sucesso!',
      titleClass: 'success',
      description: 'Aviso removido com sucesso!',
    });
    expect(wrapper.vm.notice.id).toEqual(notice.id);
  });

  test('adiciona um novo aviso', async () => {
    vi.mocked(NoticeApi.post).mockReturnValueOnce(
      Promise.resolve({
        errors: false,
      })
    );

    wrapper.vm.notice.text = 'Este é um novo aviso criado!';

    const button = wrapper.find('[type="button"]');

    button.trigger('click');

    await wrapper.vm.$nextTick();

    await flushPromises();

    expect(dialogSpy).toHaveBeenCalledWith({
      title: 'Sucesso!',
      titleClass: 'success',
      description: 'Aviso salvo com sucesso!',
    });
    expect(wrapper.vm.notice).toEqual({
      id: notice.id,
      text: 'Este é um novo aviso criado!',
    });
  });

  test('ao adicionar um novo aviso, retorna erro', async () => {
    vi.mocked(NoticeApi.post).mockRejectedValueOnce({});

    wrapper.vm.notice.text = 'Aqui vai um teste com erro desconhecido';

    const button = wrapper.find('[type="button"]');

    button.trigger('click');

    await wrapper.vm.$nextTick();

    await flushPromises();

    expect(dialogSpy).toHaveBeenCalledWith({
      title: 'Erro',
      description:
        'Não foi possível salvar o aviso. Por favor, entre em contato com o suporte.',
      titleClass: 'danger',
      iconLeft: 'status-red',
    });
  });
});
