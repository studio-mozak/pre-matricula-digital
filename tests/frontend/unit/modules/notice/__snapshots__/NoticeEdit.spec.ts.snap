// Vitest Snapshot v1

exports[`NoticeEdit > should match snapshot 1`] = `
<div>
  <!--v-if-->
  <div>
    <div
      class="row mt-5"
    >
      <div
        class="col-12 col-lg-10 offset-lg-1"
      >
        <h2
          class="title-find-school"
        >
          Avisos
        </h2>
        <div
          class="mt-4"
        >
          <p
            class="font-hind text-justify"
          >
             Ao inserir conteúdo no editor de texto abaixo a mensagem será apresentada na tela inicial de inscrições para os responsáveis na seção Avisos, respeitando a formatação definida. Use o espaço para inserir orientações aos responsáveis no processo de pré-matrícula do município. 
          </p>
        </div>
        <div
          class="mt-4"
        >
          <x-field-stub
            allowclear="false"
            bind="[object Object]"
            containerclass="col-12 p-0"
            disabled="false"
            errors="false"
            loading="false"
            modelvalue="test notice"
            name="text"
            options=""
            searchable="false"
            type="RICH_TEXT"
          />
        </div>
      </div>
    </div>
    <div
      class="mt-2 col-12 col-lg-10 offset-lg-1 p-0"
    >
      <div
        class="d-flex justify-content-end"
      >
        <button
          class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 col-12 col-md-2"
          role="button"
          tabindex="0"
          type="button"
        >
          <span
            class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
          >
            <span
              class="block"
            >
              Salvar
            </span>
          </span>
          <transition-stub
            appear="false"
            css="true"
            name="x-transition--fade"
            persisted="false"
          >
            <!---->
          </transition-stub>
        </button>
      </div>
    </div>
  </div>
</div>
`;
